src/libfetchers/fetchers.cc

```diff
@@ -412,3 +413,20 @@ std::string publicKeys_to_string(const std::vector<PublicKey>& publicKeys)
 }
 
 }
+
+namespace nlohmann {
+
+using namespace nix;
+
+fetchers::PublicKey adl_serializer<fetchers::PublicKey>::from_json(const json & json) {
+    auto type = optionalValueAt(json, "type").value_or("ssh-ed25519");
```

Comment:
Also we should not repeat the default value.

Do something along the lines of 
```suggestion
    if (auto typeOpt = optionalValueAt(json, "type"))
        res.type = *typeOpt;
```
or do a
```c++`
constexpr static std::string_view defaultType = "ssh-ed25519";
std::string type = defaultType;
```
on the class so we can use `defaultType` here.

Reply:
That makes a ton of sense, should I include that in the next PR or also separate? (I can't reopen a merged PR, right?)

Reply:
@haenoe yeah can't reopen merged. (And I don't regret merging this one! :)) but you can do in next PR.